programs/zstdcli.c

```diff
@@ -409,8 +449,20 @@ static int readSizeTFromCharChecked(const char** stringPtr, size_t* value)
         if (**stringPtr=='i') (*stringPtr)++;
         if (**stringPtr=='B') (*stringPtr)++;
     }
-    *value = result;
-    return 0;
+
+    //Only well formed entries will be 0 or , (specifically for zstd compound)
+    if (**stringPtr == 0 || **stringPtr == ',') {
+
+        *value = result;
+        return 0;
+    
+    }else{
+
+        *value = 0;
+        return 2;
```

Comment:
Please add documentation for this case.

Reply:
@yoniko - not a problem, can I quickly clarify what bit you mean?

Reply:
Function has documentation for return values, please add documentation for this new value.